src/userbase-js/src/localData.js

```diff
@@ -1,69 +1,89 @@
-  const session = { username, signedIn, sessionId, creationDate }
-  const sessionString = JSON.stringify(session)
-
-  if (rememberMe === 'local') {
-    localStorage.setItem('userbaseCurrentSession', sessionString)
-  } else if (rememberMe === 'session') {
-    sessionStorage.setItem('userbaseCurrentSession', sessionString)
+  try {
+    const session = { username, signedIn, sessionId, creationDate }
+    const sessionString = JSON.stringify(session)
+
+    if (rememberMe === 'local') {
+      localStorage.setItem('userbaseCurrentSession', sessionString)
+    } else if (rememberMe === 'session') {
+      sessionStorage.setItem('userbaseCurrentSession', sessionString)
+    }
+  } catch (e) {
+    // ok
   }
 }
 
 const getCurrentSession = () => {
-  const sessionStorageCurrentSessionString = sessionStorage.getItem('userbaseCurrentSession')
+  try {
+    const sessionStorageCurrentSessionString = sessionStorage.getItem('userbaseCurrentSession')
 
-  if (sessionStorageCurrentSessionString) {
-    const currentSession = JSON.parse(sessionStorageCurrentSessionString)
+    if (sessionStorageCurrentSessionString) {
+      const currentSession = JSON.parse(sessionStorageCurrentSessionString)
 
-    if (!currentSession.signedIn) {
-      const localCurrentSessionString = localStorage.getItem('userbaseCurrentSession')
+      if (!currentSession.signedIn) {
+        const localCurrentSessionString = localStorage.getItem('userbaseCurrentSession')
 
-      if (localCurrentSessionString) {
-        const localCurrentSession = JSON.parse(localCurrentSessionString)
+        if (localCurrentSessionString) {
+          const localCurrentSession = JSON.parse(localCurrentSessionString)
 
-        // allows session from localStorage to override sessionStorage if signed in
-        // to localStorage session and not signed in to sessionStorage session
-        if (localCurrentSession.signedIn) {
-          return {
-            ...localCurrentSession,
-            rememberMe: 'local'
+          // allows session from localStorage to override sessionStorage if signed in
+          // to localStorage session and not signed in to sessionStorage session
+          if (localCurrentSession.signedIn) {
+            return {
+              ...localCurrentSession,
+              rememberMe: 'local'
+            }
           }
         }
       }
-    }
 
-    return {
-      ...JSON.parse(sessionStorageCurrentSessionString),
-      rememberMe: 'session'
+      return {
+        ...JSON.parse(sessionStorageCurrentSessionString),
+        rememberMe: 'session'
+      }
     }
-  }
 
-  const localSessionString = localStorage.getItem('userbaseCurrentSession')
-  return localSessionString && {
-    ...JSON.parse(localSessionString),
-    rememberMe: 'local'
+    const localSessionString = localStorage.getItem('userbaseCurrentSession')
+    return localSessionString && {
+      ...JSON.parse(localSessionString),
+      rememberMe: 'local'
+    }
+  } catch (e) {
+    // ok
   }
```

Comment:
Why is it okay to swallow the exception here, and in the other places?

Reply:
(If this answer is good, let me consolidate all functions with a single try catch wrapper, and make the comment clearer in the code why it's ok before merging)

The SDK is not reliant on successful calls to any of these functions because local/sessionStorage are non-critical benefits. If they happen to be available, they're helpful, but if not, all good, the SDK functions totally fine. If none succeed, it's functionally the same as if rememberMe is 'none'.

Reply:
Sounds good. Let's refactor it then to have one handler.